### DISCUSSION TITLE
Why C# for gameplay

### DISCUSSION THREAD

**Discussion by Lewiscowles1986** [ForNeVeR/overtone#59] [Q&A]
While I understand C# for all the tooling, I don't quite understand it for the gameplay. It feels like the point and click aspects of the game could be handled via a web-ui, which would also make this title cross-os compatible, and give a wide range of supporting, common API's

Not expecting you to change direciton, more curious to see if it was ever considered?

**Comment by ForNeVeR** [comment]
First, note we don't use C# here at all. The game code is all written in F#. There were some auxiliary modules written in C#, but I've extracted them to a separate repository and made NuGet libraries (for possible reuse for other, non-game purposes).

> make this title cross-os compatible, and give a wide range of supporting, common API's

Are the things that motivate the usage of .NET for the game implementation 😅 

> It feels like the point and click aspects of the game could be handled via a web-ui

I am also considering compiling the game to WASM to run it in the browser. .NET is not an obstacle in that regard at all.

But I also have to admit that I am biased towards all things .NET, and an implementation using alternate technologies would not motivate me to participate so much as this one.

But regardless of that, aside from implementing the game in this repository, I am also trying to collect the documentation in a form that would be digestible by other people. I would be thrilled to know that someone else is working on an alternative using other technologies, using the documentation on the game data formats collected here (or elsewhere).

So, to answer your question directly: the game implementation that would run in the browser, while is not the first target, is at least considered. Getting rid of .NET, on the other hand, is not considered.
